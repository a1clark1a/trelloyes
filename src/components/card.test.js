import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Card from "./Card";
import STORE from "../store";

describe("Card component", () => {
  //smoke test
  const allCards = Object.values(STORE.allCards);
  const cards = allCards.map(elem => {
    return <Card key={elem.id} title={elem.title} content={elem.content} />;
  });

  it("Renders without crashing Test", () => {
    const div = document.createElement("div");
    ReactDOM.render([cards], div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //Snapshot test
  it("UI renders", () => {
    const tree = renderer.create([cards]).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
