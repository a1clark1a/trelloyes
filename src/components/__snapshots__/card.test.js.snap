// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component UI renders 1`] = `
Array [
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      First card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Second card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Third card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Fourth card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Fifth card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Sixth card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Seventh card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Eighth card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Ninth card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Tenth card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Eleventh card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Twelfth card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
  <div
    className="Card"
  >
    <button
      type="button"
    >
      delete
    </button>
    <h3>
      Thirteenth card
    </h3>
    <p>
      lorem ipsum
    </p>
  </div>,
]
`;
